guikit/threads.py

```diff
@@ -121,11 +121,34 @@ def __new__(cls, window: Optional[wx.Frame] = None):
             cls._instance = object.__new__(cls)
             cls._instance._window = window
             cls._instance._workers = {}
+            cls._instance._workers_lock = threading.Lock()
+
+            cls._instance._window.Bind(
```

Comment:
This doesn't feel right. Having the `ThreadPool` controlling how and when the main window is closed seems odd. I would put this functionality in here https://github.com/ImperialCollegeLondon/guikit/blob/cb60109050463b4c9ce1e5ff89f23d2c405f9200/guikit/core.py#L211

You can bind the event in the `OnInit` method and then have a separate method (the same one you have now, but on that class) stopping all threads and destroying the main window.

Reply:
Good shout! Done